torchqdynamics/mesolve.py

```diff
@@ -60,16 +63,71 @@ def forward(self, t, dt, rho):
             M0 @ rho @ M0.adjoint() + dt *
             (self.jump_ops @ rho.unsqueeze(0) @ self.jumpdag_ops).sum(dim=0)
         )
-        rho = rho / rho.trace()
+        rho = rho / rho.trace().real
         return rho
 
     def forward_adjoint(self, t, dt, phi):
         raise NotImplementedError
 
 
-class MERouchon2(MERouchon):
+class MERouchon1_5(MERouchon):
     def forward(self, t, dt, rho):
+        """Compute rho(t+dt) using a Rouchon method of order 1.5."""
+        # non-hermitian Hamiltonian at time t
+        H_nh = self.H(t) - 0.5j * self.sum_nojump
+
+        # build time-dependent Kraus operators
+        M0 = self.I - 1j * dt * H_nh
+        S = M0.adjoint() @ M0 + dt * self.sum_nojump
+        S_inv_sqrtm = self.inv_sqrtm(S)
+
+        # compute rho(t+dt)
+        rho = S_inv_sqrtm @ rho @ S_inv_sqrtm.adjoint()
```

Comment:
This is a quite un-natural way to write the normalisation, but I see why you did. Anyway this will be udpated when we specify the code for time-dep vs time-indep H.

Reply:
Yes if everything is time-independent, this is not the right way. But in the general time-dependent way it's faster.